src/RedirectMiddleware.php

```diff
@@ -98,7 +99,18 @@ public function checkRedirect(
             );
         }
 
-        return $this($nextRequest, $options);
+        /** @var PromiseInterface|ResponseInterface $response */
+        $options['_previous_uris'][] = $request->getUri();
+        $response = $this($nextRequest, $options);
+
+        // Add headers to be able to track history of redirects.
+        if (!empty($options['allow_redirects']['track_redirects'])) {
+            $response = $response->wait(TRUE);
+            if (!$response->hasHeader('X-Guzzle-Redirect')) {
```

Comment:
Well, this doesn't work as you expect it. With this if() we ensure that just the most outer response, so the one all the previous URIs available, sets the header, I think, but swapping just that lets the test fail